### PR TITLE
Update README and CONTRIBUTING guides

### PR DESCRIPTION
**Pull request by Joshua-Douglas**
This PR is primarily intended to update the repository documentation with more up to date build instructions and a more complete description of the project. I made some minor updates to the source code, but only to address docker build errors. 

Code changes:
* Increment the api dockerfile to use python 3.10, since we are using match case statements 
* Update the app "nginx.conf" to be optional, since the repo does not contain one

Documentation Changes:
* Add a CONTRIBUTING guide, and move contribution instructions out of README
* Add docker build instructions to README, and test instructions
  * Note: I could not figure out how to run the app test cases within the app container, since the container does not have `npm` or `vitest` installed (or at least these installs are not on PATH and are not accessible from `docker exec`)
* Add more complete project overview and technology overview
* The main project README now focuses on how to build all the projects together using. The individual project README focus on building each project locally, so I moved the `app` local build instructions to a new `app\README.md` file

### CODE REVIEW

File: CONTRIBUTING.md
Comment (paulespinosa) [inline CONTRIBUTING.md:37]: Is forking a new contributor policy we need to start practicing or has this always been the done?

File: CONTRIBUTING.md
Comment (agosmou) [inline CONTRIBUTING.md:27]: should we give a template on what info should include or is it better to leave open-ended? Are there any labels that are required such as story points etc.?

Example copied the below from the website team:
### Overview
REPLACE THIS TEXT - Clearly state the purpose of this issue in 2 sentences or less.  We write ours a modified user story in this format: _We need to do X for Y reason._

### Action Items
REPLACE THIS TEXT - List the research to be done, or the steps to be completed.
Note: If the steps can be divided into tasks for more than one person, we recommend dividing it up into separate issues, or assigning it as a pair programming task.

### Resources/Instructions
REPLACE THIS TEXT - Provide links to resources or instructions that may help with this issue. This can include files to be worked on, external sites with solutions, documentation, etc.

File: CONTRIBUTING.md
Comment (agosmou) [inline CONTRIBUTING.md:56]: This looks awesome! Super thorough!

File: README.md
Comment (agosmou) [inline README.md:5]: "... in their *exact..."

File: README.md
Comment (agosmou) [inline README.md:30]: i love this section! thanks for the definitions

File: README.md
Comment (paulespinosa) [inline README.md:5]: "exact policies ..."

File: README.md
Comment (paulespinosa) [inline README.md:24]: `connexion` uses the OpenAPI specification to map API URLs to specific python functions within our `Flask` app, and will handle routing, validation , security, and parameter passing when an API request is made.

File: README.md
Comment (paulespinosa) [inline README.md:23]: The HUU API is implemented using `connexion`. Let's say the API is a Python connexion application and as the bullet point, we can say we chose the variant that runs in top of Flask.

File: README.md
Comment (paulespinosa) [inline README.md:26]: If the `api` description above is updated, then this should also be updated to say "...intermediary between an HTTP proxy and the Home Unite Us API" rather than "`Flask` application".

**Review by tylerthome — APPROVED** [review]

**Comment by Joshua-Douglas** [comment]
> Would it be beneficial for us to add a line in `api/README.md` regarding the setup of virtual environments and the selection of the python interpreter in VSCode? If this is helpful, I could write something up similar to the [flask docs](https://flask.palletsprojects.com/en/2.3.x/installation/#virtual-environments)

Hey @agosmou, Great suggestion. It turns out that @paulespinosa is already adding some venv instructions as part of PR #544 so I left it out for now. 

I think in the future we can add more detailed guides for the local build instructions, but I mainly focused on the docker build in this PR.

File: CONTRIBUTING.md
Comment (Joshua-Douglas) [reply CONTRIBUTING.md]: That's right. Template shows by default and is stored in the repo. I added a link to the template in [ebfba50](https://github.com/hackforla/HomeUniteUs/pull/557/commits/ebfba503f84ed549893cd2f68623dc90597fdafe).

The template is identical to what @agosmou pasted.

File: README.md
Comment (Joshua-Douglas) [reply README.md]: Thanks! Corrected in [ebfba50](https://github.com/hackforla/HomeUniteUs/pull/557/commits/ebfba503f84ed549893cd2f68623dc90597fdafe)

File: CONTRIBUTING.md
Comment (Joshua-Douglas) [reply CONTRIBUTING.md]: Thanks! Corrected in [ebfba50](https://github.com/hackforla/HomeUniteUs/pull/557/commits/ebfba503f84ed549893cd2f68623dc90597fdafe)

File: CONTRIBUTING.md
Comment (Joshua-Douglas) [reply CONTRIBUTING.md]: Ahhh good catch. I was assuming we didn't all have write access, but it looks like the onboarding process does give us that level of access. 

Forks are not necessary, so I updated the contribution guideline to say that contributions can be directly submitted after onboarding within [ebfba50](https://github.com/hackforla/HomeUniteUs/pull/557/commits/ebfba503f84ed549893cd2f68623dc90597fdafe).

Also mentioned that the dev should assign themselves to the issue they plan on working.

File: README.md
Comment (Joshua-Douglas) [reply README.md]: Thanks! Corrected in [ebfba50](https://github.com/hackforla/HomeUniteUs/pull/557/commits/ebfba503f84ed549893cd2f68623dc90597fdafe)

File: README.md
Comment (Joshua-Douglas) [reply README.md]: Thanks! Added links and more complete description in [ebfba50](https://github.com/hackforla/HomeUniteUs/pull/557/commits/ebfba503f84ed549893cd2f68623dc90597fdafe)

File: README.md
Comment (Joshua-Douglas) [reply README.md]: Great clarification. So true. Corrected in [ebfba50](https://github.com/hackforla/HomeUniteUs/pull/557/commits/ebfba503f84ed549893cd2f68623dc90597fdafe)

File: README.md
Comment (Joshua-Douglas) [reply README.md]: Perfect! Yeah I was intentionally vague because it wasn't clear in my head. Thanks for the help.

Comment (Joshua-Douglas) [reply README.md]: Corrected in [ebfba50](https://github.com/hackforla/HomeUniteUs/pull/557/commits/ebfba503f84ed549893cd2f68623dc90597fdafe)

**Comment by Joshua-Douglas** [comment]
Thanks @tylerthome, @agosmou, and @paulespinosa for the review. 

I believe I've addressed everyone's comments so we should be good to merge unless anyone else has additional comments.

**Comment by paulespinosa** [comment]
This is very helpful. Thank you.

**Pull request merged**
